internal/gitpartitionsync/producer/tar.go

```diff
@@ -61,19 +60,22 @@ func (g *GitPartitionSyncProducer) tarRepos(repoPath string, sync syncConfig) (s
 		}
 
 		// open files for taring
-		f, err := os.Open(file)
+		f, err := os.Open(filepath.Clean(file))
 		if err != nil {
 			return err
 		}
+		// Closing a healthy file twice yields a syscall.EINVAL
+		// error which is safe to discard in this case.
+		defer func() { _ = f.Close() }()
```

Comment:
Can't we simply do `defer f.Close()` here?

Reply:
We can.

- - -
**Side note:**

This is a pattern some [projects] are picking up on. The aim here is to show that you explicitly ignore the return value with, I suppose, all the consequences of doing it. Some stricter [linters] also pick on this and shout about it, especially for operations on *so.File, where if the file is open for writing, ignoring errors can lead to issues.

This style, let it be practical or perhaps even philosophical in approach, since currently ignoring error values is allowed within Go specification, seems to have been trying to turn itself into a de facto standard per:

- [proposal: spec: require return values to be explicitly used or ignored (Go 2)](https://github.com/golang/go/issues/20803)

[projects]: https://github.com/thanos-io/thanos/blob/main/docs/contributing/coding-style-guide.md#defers-dont-forget-to-check-returned-errors
[linters]: https://github.com/kisielk/errcheck

Reply:
Done. See: 6dfb25e